rdsbroker/broker.go

```diff
@@ -391,10 +391,57 @@ func (b *RDSBroker) LastOperation(instanceID string) (brokerapi.LastOperationRes
 	return lastOperationResponse, nil
 }
 
+func (b *RDSBroker) CheckAndRotateCredentials() error {
+	dbInstanceDetailsList, err := b.dbInstance.DescribeByTag("Broker Name", b.brokerName)
+	if err != nil {
+		b.logger.Error("Could not retrieve the list of DB instances", err)
+		return err
+	}
+
+	for _, dbInstanceDetails := range dbInstanceDetailsList {
+		dbAddress := dbInstanceDetails.Address
+		dbPort := dbInstanceDetails.Port
+		masterUsername := dbInstanceDetails.MasterUsername
+		serviceInstanceID := b.dbInstanceIdentifierToServiceInstanceID(dbInstanceDetails.Identifier)
+
+		var dbName string
+		if dbInstanceDetails.DBName != "" {
+			dbName = dbInstanceDetails.DBName
+		} else {
+			dbName = b.dbName(dbInstanceDetails.Identifier)
+		}
+
+		sqlEngine, err := b.sqlProvider.GetSQLEngine(dbInstanceDetails.Engine)
+		if err != nil {
+			b.logger.Error("Could not get the SQL Engine", err)
+			return err
+		}
+
+		err = sqlEngine.Open(dbAddress, dbPort, dbName, masterUsername, b.masterPassword(serviceInstanceID))
+		if err != nil {
+			b.logger.Error(fmt.Sprintf("Can't connect to DB %v at %v, will attempt to reset the password", dbName, dbAddress), err)
+			dbInstanceDetails.MasterUserPassword = b.masterPassword(serviceInstanceID)
+			err = b.dbInstance.Modify(dbInstanceDetails.Identifier, *dbInstanceDetails, true)
+			if err != nil {
+				b.logger.Error("Could not reset the master password", err)
+				return err
+			}
+		} else {
+			defer sqlEngine.Close()
```

Comment:
It would be good to move this just after `sqlEngine.Open()` so it's more readable that we care about closing connections.